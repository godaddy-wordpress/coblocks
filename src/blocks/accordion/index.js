/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
