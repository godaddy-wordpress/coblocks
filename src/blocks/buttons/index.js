/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
