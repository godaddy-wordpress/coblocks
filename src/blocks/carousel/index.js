/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
