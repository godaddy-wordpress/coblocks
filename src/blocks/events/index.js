/**
 * External dependencies
 */
import { EventsIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies.
 */
import deprecated from './deprecated';
import edit from './edit';
import example from './example';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies.
 */
import { Icon } from '@wordpress/components';
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	deprecated,
	description: __( 'Add a list of events or display events from a public calendar.', 'coblocks' ),
	edit,
	example,
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'calendar', 'coblocks' ),
		/* translators: block keyword */
		__( 'date', 'coblocks' ),
	],
	save,
	supports: {
		align: [ 'wide', 'full' ],
	},
	title: _x( 'Events', 'block name', 'coblocks' ),
	transforms,
};

export { metadata, name, category, settings };
