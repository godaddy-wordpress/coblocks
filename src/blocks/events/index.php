<?php
/**
 * Server-side rendering of the `coblocks/events` block.
 *
 * @package CoBlocks
 */

/**
 * Renders the `events` block on server.
 *
 * @param array  $attributes The block attributes.
 * @param string $content    The post content.
 *
 * @return string Returns the events content.
 */
function coblocks_render_events_block( $attributes, $content ) {

	if ( empty( $attributes['externalCalendarUrl'] ) ) {
		return $content;
	}

	try {
		$ical = new \CoBlocks_ICal(
			false,
			// Default values.
			array(
				'default_span'                  => 2,
				'default_time_zone'             => 'UTC',
				'default_week_start'            => 'MO',
				'disable_character_replacement' => false,
				'filter_days_after'             => null,
				'filter_days_before'            => null,
				'skip_recurrence'               => false,
				'use_timezone_with_r_rules'     => false,
			)
		);
		$ical->init_url( $attributes['externalCalendarUrl'] );

		if ( 'all' === $attributes['eventsRange'] ) {
			$events = $ical->events_from_range();
		} else {
			$events = $ical->events_from_interval( $attributes['eventsRange'] );
		}
		// Limit to 100 events.
		$events = array_slice( $events, 0, 100 );

		$text_color_class  = is_array( $attributes ) && isset( $attributes['textColor'] ) ? "has-{$attributes['textColor']}-color" : false;
		$custom_text_color = is_array( $attributes ) && isset( $attributes['customTextColor'] ) && isset( $attributes['hasColors'] ) && ( ! $attributes['hasColors'] && ! isset( $attributes['textColor'] ) ) ? "color: {$attributes['customTextColor']};" : '';

		$class = 'wp-block-coblocks-events';

		if ( true === $attributes['showCarousel'] ) {
				$class .= ' swiper-wrapper-loading';
		}

		if ( isset( $attributes['className'] ) ) {
			$class .= ' ' . $attributes['className'];
		}

		if ( isset( $attributes['align'] ) ) {
			$class .= ' align' . $attributes['align'];
		}

		$events_layout = '<div class="wp-block-coblocks-events wp-block-coblocks-events-front-container">';

		$events_layout .= sprintf(
			'<div class="swiper-container-external wp-block-coblocks-front-events-swiper-container"><div class="%1$s" data-per-page="%2$s">',
			esc_attr( $class ),
			esc_attr( $attributes['eventsToShow'] )
		);

		foreach ( $events as $event ) {
			$events_layout .= '<div class="wp-block-coblocks-event-item swiper-slide">';

			$dtstart           = $ical->ical_date_to_date_time( $event->dtstart_array[3] );
			$dtend             = $ical->ical_date_to_date_time( $event->dtend_array[3] );
			$start_date_string = strtotime( $dtstart->format( 'YmdHis' ) );
			$end_date_string   = strtotime( $dtend->format( 'YmdHis' ) );
			$year              = gmdate( 'Y', $start_date_string );
			$month             = gmdate( 'F', $start_date_string );
			$start_date        = gmdate( 'd', $start_date_string );
			$end_date          = gmdate( 'd', $end_date_string );
			$start_time        = gmdate( 'c', $start_date_string );
			$end_time          = gmdate( 'c', $end_date_string );
			$time_string       = $start_time . ' - ' . $end_time;
			$title             = $event->summary;
			$description       = $event->description;
			$location          = $event->location;
			$event_duration    = $end_date_string - $start_date_string;
			$is_timed_event    = str_contains( $event->dtstart, 'Z' );
			$one_day           = 86400;

			if ( $event_duration === $one_day && ! $is_timed_event ) {

				$event_time_string = sprintf(
					'<span class="wp-block-coblocks-events__time">%1$s - %2$s</span>',
					gmdate( 'g:ia', $start_date_string ),
					gmdate( 'g:ia', $end_date_string )
				);

				$events_layout .= coblocks_render_single_day_event_item(
					$start_date,
					$month,
					$year,
					$title,
					$description,
					$event_time_string,
					$location
				);

			} elseif ( $event_duration > $one_day && ! $is_timed_event ) {

				$event_time_string = sprintf(
					'<span class="wp-block-coblocks-events__time">%1$s - %2$s</span>',
					gmdate( 'g:ia', $start_date_string ),
					gmdate( 'g:ia', $end_date_string )
				);

				$events_layout .= coblocks_render_multi_day_event_item(
					$start_date,
					$end_date - 1,
					$month,
					$year,
					$title,
					$description,
					$event_time_string,
					$location
				);

			} elseif ( $event_duration < $one_day && $is_timed_event ) {

				$event_time_string = sprintf(
					'<span data-start-time=%1$s data-end-time=%2$s class="wp-block-coblocks-events__time wp-block-coblocks-events__time-formatted"></span>',
					gmdate( 'c', $start_date_string ),
					gmdate( 'c', $end_date_string )
				);

				$events_layout .= coblocks_render_single_day_event_item(
					$start_date,
					$month,
					$year,
					$title,
					$description,
					$event_time_string,
					$location
				);

			} elseif ( $event_duration > $one_day || $event_duration === $one_day && $is_timed_event ) {

				$event_time_string = sprintf(
					'<span data-start-time=%1$s data-end-time=%2$s class="wp-block-coblocks-events__time wp-block-coblocks-events__time-formatted"></span>',
					gmdate( 'c', $start_date_string ),
					gmdate( 'c', $end_date_string )
				);

				$events_layout .= coblocks_render_multi_day_event_item(
					$start_date,
					$end_date,
					$month,
					$year,
					$title,
					$description,
					$event_time_string,
					$location
				);

			}

			$events_layout .= '</div>';
		}

		$events_layout .= '</div>';

		$events_layout .= '<button class="wp-coblocks-events-nav-button__prev" id="wp-coblocks-event-swiper-prev" style="visibility: hidden" />';
		$events_layout .= '<button class="wp-coblocks-events-nav-button__next" id="wp-coblocks-event-swiper-next" style="visibility: hidden" />';

		$events_layout .= '</div>';

		$events_layout .= '</div>';

		return $events_layout;

	} catch ( \Exception $e ) {

		return '<div class="components-placeholder"><div class="notice notice-error">' . __( 'An error has occurred, check for calendar privileges to make sure it is public or try again later.', 'coblocks' ) . '</div></div>';

	}
}

/**
 * Formats a generic event item into HTML
 *
 * @param string $date_range the range of dates for the event.
 * @param string $month month of the event.
 * @param string $year year of the event.
 * @param string $title title of event.
 * @param string $description string description of event.
 * @param string $time_string the time range of the event.
 * @param string $location location of the event.
 */
function coblocks_render_event_item(
	$date_range,
	$month,
	$year,
	$title,
	$description,
	$time_string,
	$location
) {

	$event_layout = sprintf(
		'
		<div class="wp-block-coblocks-events__date">
			<span class="wp-block-coblocks-events__day">%1s</span>
			<div>
				<span class="wp-block-coblocks-events__month">%2$s</span>
				<span class="wp-block-coblocks-events__year">%3$s</span>
			</div>
		</div>',
		$date_range,
		$month,
		$year
	);

	$event_layout .= sprintf(
		'<div class="wp-block-coblocks-events__content">
			<span class="wp-block-coblocks-events__title">%1$s</span>
			<span class="wp-block-coblocks-events__description">%2$s</span>
		</div>',
		$title,
		$description
	);

	$event_layout .= sprintf(
		'<div class="wp-block-coblocks-events__details">
			%1$s
			<span class="wp-block-coblocks-events__location">%2$s</span>
		</div>',
		$time_string,
		$location
	);

	return $event_layout;
}

/**
 * Formats a multi-day event into HTML
 *
 * @param string $start_date start date of event.
 * @param string $end_date end date of event.
 * @param mixed  ...$event_data list of event data arguments.
 */
function coblocks_render_multi_day_event_item( $start_date, $end_date, ...$event_data ) {
	$date_range  = $start_date;
	$date_range .= ' - ';
	$date_range .= $end_date;

	return coblocks_render_event_item( $date_range, ...$event_data );
}

/**
 * Formats a single-day event into HTML
 *
 * @param mixed ...$event_data event data.
 */
function coblocks_render_single_day_event_item( ...$event_data ) {
	return coblocks_render_event_item( ...$event_data );
}

/**
 * Registers the `events` block on server.
 */
function coblocks_register_events_block() {

	// Return early if this function does not exist.
	if ( ! function_exists( 'register_block_type' ) ) {
		return;
	}

	// Load attributes from block.json.
	ob_start();
	include COBLOCKS_PLUGIN_DIR . 'src/blocks/events/block.json';
	$metadata = json_decode( ob_get_clean(), true );

	register_block_type(
		'coblocks/events',
		array(
			'attributes'      => $metadata['attributes'],
			'render_callback' => 'coblocks_render_events_block',
		)
	);
}

add_action( 'init', 'coblocks_register_events_block' );
