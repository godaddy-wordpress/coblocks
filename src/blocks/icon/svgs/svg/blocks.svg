<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>