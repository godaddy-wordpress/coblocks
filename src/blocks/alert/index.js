/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
