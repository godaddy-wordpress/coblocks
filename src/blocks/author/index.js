/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
