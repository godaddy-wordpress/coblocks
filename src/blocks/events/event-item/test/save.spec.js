/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { createBlock, registerBlockType, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

const DEFAULT_ATTRIBUTES = {
	description: 'Some description',
	eventDay: '25',
	eventLocation: 'Somewhere',
	eventMonth: 'Nov',
	eventTime: '5:25',
	eventYear: '2099',
	title: 'Some title',
};

describe( 'coblocks/event-item', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with content', () => {
		block.attributes = DEFAULT_ATTRIBUTES;

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		// Disable reason : value must be the second parameter
		// eslint-disable-next-line no-unused-vars
		Object.entries( DEFAULT_ATTRIBUTES ).forEach( ( [ key, value ] ) => expect( serializedBlock ).toContain( value ) );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render text color', () => {
		block.attributes = DEFAULT_ATTRIBUTES;
		block.attributes.textColor = '#333333';

		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-333333-color' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
