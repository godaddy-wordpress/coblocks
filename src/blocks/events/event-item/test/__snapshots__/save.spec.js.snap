// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/event-item should render 1`] = `"<!-- wp:coblocks/event-item /-->"`;

exports[`coblocks/event-item should render text color 1`] = `
"<!-- wp:coblocks/event-item {\\"textColor\\":\\"#333333\\"} -->
<div class=\\"wp-block-coblocks-event-item swiper-slide has-333333-color has-text-color\\" data-page=\\"undefined\\"><div class=\\"wp-block-coblocks-events__date\\"><span class=\\"wp-block-coblocks-events__day\\">25</span><div><span class=\\"wp-block-coblocks-events__month\\">Nov</span><span class=\\"wp-block-coblocks-events__year\\">2099</span></div></div><div class=\\"wp-block-coblocks-events__content\\"><span class=\\"wp-block-coblocks-events__title\\">Some title</span><span class=\\"wp-block-coblocks-events__description\\" itemprop=\\"description\\">Some description</span></div><div class=\\"wp-block-coblocks-events__details\\"><span class=\\"wp-block-coblocks-events__time\\">5:25</span><span class=\\"wp-block-coblocks-events__location\\">Somewhere</span></div></div>
<!-- /wp:coblocks/event-item -->"
`;

exports[`coblocks/event-item should render with content 1`] = `
"<!-- wp:coblocks/event-item -->
<div class=\\"wp-block-coblocks-event-item swiper-slide\\" data-page=\\"undefined\\"><div class=\\"wp-block-coblocks-events__date\\"><span class=\\"wp-block-coblocks-events__day\\">25</span><div><span class=\\"wp-block-coblocks-events__month\\">Nov</span><span class=\\"wp-block-coblocks-events__year\\">2099</span></div></div><div class=\\"wp-block-coblocks-events__content\\"><span class=\\"wp-block-coblocks-events__title\\">Some title</span><span class=\\"wp-block-coblocks-events__description\\" itemprop=\\"description\\">Some description</span></div><div class=\\"wp-block-coblocks-events__details\\"><span class=\\"wp-block-coblocks-events__time\\">5:25</span><span class=\\"wp-block-coblocks-events__location\\">Somewhere</span></div></div>
<!-- /wp:coblocks/event-item -->"
`;
